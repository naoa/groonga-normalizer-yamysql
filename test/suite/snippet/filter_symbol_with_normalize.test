register normalizers/yamysql

table_create remove_phrases TABLE_PAT_KEY ShortText
load --table remove_phrases
[
{"_key": "<remove_symbol>"}
]

normalize NormalizerYaMySQL "データ+ベース" WITH_TYPES|WITH_CHECKS

table_create Entries TABLE_NO_KEY
column_create Entries content COLUMN_SCALAR ShortText

table_create Tokens TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram --normalizer NormalizerYaMySQL
column_create Tokens entries_content COLUMN_INDEX|WITH_POSITION Entries content

load --table Entries
[
{"content": "データ+ベース"}
]

select Entries \
  --output_columns 'snippet(content, \
  "データ-ベース",  "<span class=\\"w1\\">", "</span>", {"normalizer": "NormalizerYaMySQL"}), content' \
  --command_version 2 \
  --match_columns 'content' \
  --query "データ-ベース"
