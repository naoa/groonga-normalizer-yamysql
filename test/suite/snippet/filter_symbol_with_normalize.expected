register normalizers/yamysql
[[0,0.0,0.0],true]
table_create remove_phrases TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
load --table remove_phrases
[
{"_key": "<remove_symbol>"}
]
[[0,0.0,0.0],1]
normalize NormalizerYaMySQLRemovePhrase "データ+ベース" WITH_TYPES|WITH_CHECKS
[
  [
    0,
    0.0,
    0.0
  ],
  {
    "normalized": "データベース",
    "types": [
      "katakana",
      "katakana",
      "katakana",
      "katakana",
      "katakana",
      "katakana"
    ],
    "checks": [
      3,
      0,
      0,
      3,
      0,
      0,
      3,
      0,
      0,
      4,
      0,
      0,
      3,
      0,
      0,
      3,
      0,
      0
    ]
  }
]
table_create Entries TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Entries content COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
table_create Tokens TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram --normalizer NormalizerYaMySQL
[[0,0.0,0.0],true]
column_create Tokens entries_content COLUMN_INDEX|WITH_POSITION Entries content
[[0,0.0,0.0],true]
load --table Entries
[
{"content": "データ+ベース"}
]
[[0,0.0,0.0],1]
select Entries   --output_columns 'snippet(content,   "データ-ベース",  "<span class=\\"w1\\">", "</span>", {"normalizer": "NormalizerYaMySQLRemovePhrase"}), content'   --command_version 2   --match_columns 'content'   --query "データ-ベース"
[[0,0.0,0.0],[[[0],[["snippet",null],["content","ShortText"]]]]]
